gef-ui/server/controllers/unissued-facilities/index.js

```diff
@@ -0,0 +1,516 @@
+    coverStartDateMonth: coverStartDate ? format(coverStartDate, 'M') : null,
+    coverStartDateYear: coverStartDate ? format(coverStartDate, 'yyyy') : null,
+    coverEndDateDay: coverEndDate ? format(coverEndDate, 'd') : null,
+    coverEndDateMonth: coverEndDate ? format(coverEndDate, 'M') : null,
+    coverEndDateYear: coverEndDate ? format(coverEndDate, 'yyyy') : null,
+    facilityTypeString,
+    dealId,
+    facilityId,
+    status,
+    change,
+  };
+
+  return body;
+};
+
+// when changing unissued facility from unissued facility list
+const changeUnissuedAboutFacility = async (req, res) => {
+  const { params, query } = req;
+
+  try {
+    const {
+      facilityType,
+      facilityName,
+      hasBeenIssued,
+      monthsOfCover,
+      shouldCoverStartOnSubmission,
+      issueDateDay,
+      issueDateMonth,
+      issueDateYear,
+      coverStartDateDay,
+      coverStartDateMonth,
+      coverStartDateYear,
+      coverEndDateDay,
+      coverEndDateMonth,
+      coverEndDateYear,
+      facilityTypeString,
+      dealId,
+      facilityId,
+      status,
+      change,
+    } = await renderChangeFacilityPartial(params, query, false);
+    return res.render('partials/unissued-change-about-facility.njk', {
+      facilityType,
+      facilityName,
+      hasBeenIssued,
+      monthsOfCover,
+      shouldCoverStartOnSubmission,
+      issueDateDay,
+      issueDateMonth,
+      issueDateYear,
+      coverStartDateDay,
+      coverStartDateMonth,
+      coverStartDateYear,
+      coverEndDateDay,
+      coverEndDateMonth,
+      coverEndDateYear,
+      facilityTypeString,
+      dealId,
+      facilityId,
+      status,
+      change,
+    });
+  } catch (err) {
+    return res.render('partials/problem-with-service.njk');
+  }
+};
+
+// when changing unissued facility from application preview page
+const changeUnissuedAboutFacilityChange = async (req, res) => {
+  const { params, query } = req;
+
+  try {
+    const {
+      facilityType,
+      facilityName,
+      hasBeenIssued,
+      monthsOfCover,
+      shouldCoverStartOnSubmission,
+      issueDateDay,
+      issueDateMonth,
+      issueDateYear,
+      coverStartDateDay,
+      coverStartDateMonth,
+      coverStartDateYear,
+      coverEndDateDay,
+      coverEndDateMonth,
+      coverEndDateYear,
+      facilityTypeString,
+      dealId,
+      facilityId,
+      status,
+      change,
+    } = await renderChangeFacilityPartial(params, query, true);
+    return res.render('partials/unissued-change-about-facility.njk', {
+      facilityType,
+      facilityName,
+      hasBeenIssued,
+      monthsOfCover,
+      shouldCoverStartOnSubmission,
+      issueDateDay,
+      issueDateMonth,
+      issueDateYear,
+      coverStartDateDay,
+      coverStartDateMonth,
+      coverStartDateYear,
+      coverEndDateDay,
+      coverEndDateMonth,
+      coverEndDateYear,
+      facilityTypeString,
+      dealId,
+      facilityId,
+      status,
+      change,
```

Comment:
Lots of duplication here in destructuring - could we maybe create a function for this and use in both places? Something like this

```js
const something = (facility => ({
  facilityType: facility.facilityType,
  facilityName: facility.facilityName,
  hasBeenIssued: facility.hasBeenIssued,
});

...

res.render('template1.njk', something(facility));
res.render('template2.njk', something(facility));
```

Reply:
I did try something similar to this but it broke everything so I changed it to this mess.  I can have a go again?

Reply:
done